src/plugins/region_map/public/region_map_visualization.js

```diff
@@ -90,29 +93,60 @@ export function createRegionMapVisualization({
       this._kibanaMap.useUiStateFromVisualization(this._vis);
     }
 
+    async _loadConfig(fileLayerConfig) {
+      // Load the selected layer from the metadata-service.
+      // Do not use the selectedLayer from the visState.
+      // These settings are stored in the URL and can be used to inject dirty display content.
+
+      if (fileLayerConfig.isEMS) {
+        //Hosted by EMS. Metadata needs to be resolved through EMS.
+        return await serviceSettings.loadFileLayerConfig(fileLayerConfig);
+      }
+
+      if (fileLayerConfig.layerId && fileLayerConfig.layerId.startsWith(`${ORIGIN.EMS}.`)) {
+        //fallback for older saved objects
+        return await serviceSettings.loadFileLayerConfig(fileLayerConfig);
+      }
+
+      //Configured in the kibana.yml. Needs to be resolved through the settings.
+      const configuredLayer = regionmapsConfig.layers.find(
+        (layer) => layer.name === fileLayerConfig.name
+      );
+
+      if (configuredLayer) {
+        return {
+          ...configuredLayer,
+          attribution: _.escape(configuredLayer.attribution ? configuredLayer.attribution : ''),
+        };
+      }
+
+      return null;
+    }
+
     async _updateParams() {
       await super._updateParams();
-      const visParams = this._params;
 
-      if (!visParams.selectedJoinField && visParams.selectedLayer) {
-        visParams.selectedJoinField = visParams.selectedLayer.fields[0];
+      const selectedLayer = await this._loadConfig(this._params.selectedLayer);
+
+      if (!this._params.selectedJoinField && selectedLayer) {
+        this._params.selectedJoinField = selectedLayer.fields[0];
       }
 
-      if (!visParams.selectedJoinField || !visParams.selectedLayer) {
+      if (!this._params.selectedJoinField || !selectedLayer) {
         return;
```

Comment:
While we're in here, maybe we should just set this to explicitly `return null;`

Reply:
hmmm... I wouldn't make changes to this method signature in a spot-fix. `_updateParams` has a void-return, and the response is not handled by any of the callers.